nowcast/delphi_nowcast/data_containers.py

```diff
@@ -0,0 +1,95 @@
+"""Data container classes for holding sensor configurations and data needed for fusion."""
+
+from dataclasses import dataclass, field
+from typing import List, Union
+
+from numpy import ndarray, nan, nanmean, isnan
+from pandas import date_range
+
+
+@dataclass(frozen=True)
+class SensorConfig:
+    """Dataclass for specifying a sensor's name, number of lag days, and origin source/signal."""
+
+    source: str
+    signal: str
+    name: str
+    lag: int
+
+
+@dataclass
+class LocationSeries:
+    """Data class for holding time series data specific to a single location."""
+
+    geo_value: str = None
+    geo_type: str = None
+    dates: List[int] = field(default_factory=lambda: [])
+    values: Union[List, ndarray] = field(default_factory=lambda: [])
+
+    def __post_init__(self):
+        """Input validation."""
+        if (self.dates is None and self.values is not None) or \
+                (self.dates is not None and self.values is None):
+            raise ValueError("Must init with both date and values.")
+
+        if (self.dates is not None and self.values is not None) and \
+                (len(self.dates) != len(self.values)):
+            raise ValueError("Length of dates and values differs.")
+
+        if len(set(self.dates)) < len(self.dates):
+            raise ValueError("Duplicate dates not allowed.")
+
+    def add_data(self, date, value):
+        """Append a date and value to existing attributes.
+
+        Safer than appending individually since the two lists shouldn't have different lengths.
+        """
+        self.dates.append(date)
+        self.values.append(value)
+
+    @property
+    def empty(self):
+        """Check if there is no stored data in the class."""
+        return not self.dates and not self.values
+
+    def get_value(self, date: int) -> float:
```

Comment:
Conceptually, aren't we just implementing a date-to-value dictionary here?

Reply:
yea basically, with a few other keys involved. maybe a better way to implement this is to store them as a dict and retrieve them with .keys() and .values() if you want them individually.

Reply:
Just pushed a new implementation where the values are stored as a day:val dict, let me know what you think.